lib/MirrorCache/Utils.pm

```diff
@@ -31,7 +31,19 @@ sub random_string {
     my ($length, $chars) = @_;
     $length //= 16;
     $chars  //= ['a' .. 'z', 'A' .. 'Z', '0' .. '9', '_'];
-    return join('', map { $chars->[rand @$chars] } 1 .. $length);
+    my $upper = scalar(@$chars);
+    return join('', map { $chars->[random_int_safe($upper)] } 1 .. $length);
+}
+
+sub random_int_safe {
+    my ($upper) = @_;
+    my $length = 4;
```

Comment:
You could document that $upper should be below 1000000000 for a decent probability distribution. In case this function gets reused in other contexts.

Reply:
Why is that? In my understanding random 4 bytes from /dev/urandom should produce random int value (below 4294967295). Why numbers below 1000000000 would be special?